Tests/PromiseTests.cs

```diff
@@ -1424,5 +1426,35 @@ public void rejected_reject_callback_is_caught_by_chained_catch()
 
             Assert.Equal(expectedException, actualException);
         }
+        
+        [Fact]
+        public void resolved_callback_is_caught_by_catch_returning_promise() {
+            var promise = new Promise<int>();
+            int excepectedValue = 1;
+            int actualValue = 0;
+
+            promise.Catch(err => Promise<int>.Resolved(-1))
+                   .Then(result => { actualValue = result; })
+                   .Catch(err => throw new Exception("Should not happend."));
+            
+            promise.Resolve(1);
```

Comment:
it would make more sense here to make the first promise reject and verify that `actualValue` was set to -1, to verify that the value from a promise returned inside a `Catch` is coming through.